tests/controllers/articles.test.ts

```diff
@@ -0,0 +1,248 @@
+        expect(typeof article.slug).toBe('string');
+        expect(article.posted_on).toBeInstanceOf(Date);
+        expect(typeof article.article_image_url).toBe('string');
+        expect(typeof article.summary).toBe('string');
+        expect(typeof article.author_name).toBe('string');
+        expect(typeof article.author_role).toBe('string');
+        expect(typeof article.author_image_url).toBe('string');
+        expect(typeof article.html_content).toBe('string');
+        expect(typeof article.category_name).toBe('string');
+        expect(typeof article.category_slug).toBe('string');
+        expect(typeof article.reading_time).toBe('number');
+    });
+
+    test('generateRelatedArticles should create valid objects', () => {
+        const testId = 1;
+        const testRelatedIds = [2, 3, 4];
+        const expectedResponse = [
+            { article_id: 1, related_article_id: 2 },
+            { article_id: 1, related_article_id: 3 },
+            { article_id: 1, related_article_id: 4 },
+        ];
+        const response = generateRelatedArticles(testId, testRelatedIds);
+        expect(response).toEqual(expectedResponse);
+    });
+
+    test('addToRelatedArticlesTable cant add empty array', async () => {
+        expect.assertions(1);
+        const emptyArray = [];
+        const response = await addToRelatedArticlesTable(blog, 1, emptyArray);
+        expect(response).toEqual(emptyArray);
+    });
+
+    test('addToRelatedArticlesTable works with undefined', async () => {
+        expect.assertions(1);
+        const emptyArray = [];
+        const response = await addToRelatedArticlesTable(blog, 1);
+        expect(response).toEqual(emptyArray);
+    });
+
+    test('addArticle should add an article correctly', async () => {
+        expect.assertions(15);
+        const addedArticle = await addArticle(blog, newArticle);
+        expect(typeof addedArticle).toBe('object');
+        expect(typeof addedArticle.id).toBe('number');
+        expect(typeof addedArticle.title).toBe('string');
+        expect(typeof addedArticle.subtitle).toBe('string');
+        expect(typeof addedArticle.slug).toBe('string');
+        expect(addedArticle.posted_on).toBeInstanceOf(Date);
+        expect(typeof addedArticle.article_image_url).toBe('string');
+        expect(typeof addedArticle.summary).toBe('string');
+        expect(typeof addedArticle.author_name).toBe('string');
+        expect(typeof addedArticle.author_role).toBe('string');
+        expect(typeof addedArticle.author_image_url).toBe('string');
+        expect(typeof addedArticle.html_content).toBe('string');
+        expect(typeof addedArticle.category_name).toBe('string');
+        expect(typeof addedArticle.category_slug).toBe('string');
+        expect(typeof addedArticle.reading_time).toBe('number');
+    });
+
+    // test('addArticle should work with custom posted_on', async () => { TODO:
+    // });
+
+    // TODO: Fix
```

Comment:
It is `xtest`, meaning the test will be skipped. I don't want to remove the code as the tests are complete, but the production code is not. But I don't want the test suite to fail just because of this, that's why it's xtested